File: LinkedList/linkedlist.rb
class Node
	def initialize(data, nextNode = nil)
		@data = data
		@next = nextNode
	end
end

class LinkedList
  def initialize
    @head = nil
  end

  def insertAtBeginnig(data)
    @head = Node.new(data, @head)
  end

  def insertAtEnd(data)
    newNode = Node.new(data)

    if @head.blank?
      @head = newNode
    end

    tail = @head
  end
end

list = LinkedList.new()
list.insertAtBeginnig('a')
list.insertAtBeginnig('b')
list.insertAtBeginnig('c')
list.insertAtBeginnig('d')
p list

File: LinkedList/linkedList.php
<?php

class Entity {
    public $date;
    public $info;

    public function __construct($date, $info) {
        $this->date = $date;
        $this->info = $info;
    }
}

class Node {
    public $data;
    public $next;
}

class LinkedList {
    public $head;

    public function __construct() {
        $this->head = null;
    }

    public function instartAtBeginning($newElement) {
        $newNode = new Node();
        $newNode->data = $newElement;
        $newNode->next = null;

        if($this->head == null) {
            $this->head = $newNode;
        } else {
            $temp = new Node();
            $temp = $this->head;

            while($temp->head != null) {
                $temp = $temp->next;
            }

            $temp->next = $newNode;
        }
    }
}

/**
 * Usage
 * $MyList = new LinkedList();
 *
 * $MyList->push_back(new Entity('2020-12-12', array("key" => "value1")));
 * $MyList->push_back(new Entity('2020-12-13', array("key" => "value2")));
 * $MyList->push_back(new Entity('2020-12-14', array("key" => "value3" ));
 **/
